File: python/incometax/contributions/philhealth_contribution.py
from .contribution import Contribution

class PhilHealthContribution(Contribution):
    """
    Calculates the PhilHealth monthly contribution depending
    on the user monthly income.
    Uses the PhilHealth Contribution Table 2019 for the computation

    """

    def __init__(self, monthly_salary = 0, membership = 'employed'):
        """
        membership is either "employed" or "self-employed"
        """
        super().__init__(monthly_salary)
        self.model = 'PhilHealth'
        self.validate_membership(membership)
        self.monthly_contribution = self.compute(monthly_salary, membership)

    def validate_membership(self, membership):
        if membership not in ('employed', 'self-employed'):
            raise ValueError({'membership': 'invalid membership'})
        self.membership = membership
    
    def compute(self, monthly_salary, membership = 'employed'):
        """
        accepts the monthly_salary and membership type
        and returns the monthly contribution

        """
        if membership == 'employed':
            if monthly_salary <= 10000:
                return 137.50
            elif monthly_salary >= 10000.01 and monthly_salary <= 39999.99:
                return ( monthly_salary * 0.0275 ) / 2
            elif monthly_salary >= 40000:
                return 1100.0
        elif membership == 'self-employed':
            if monthly_salary <= 25000:
                return 200.0
            elif monthly_salary > 25000:
                return 300.0

File: java/incometax/contribution/PhilHealthContribution.java
package incometax.contribution;

public class PhilHealthContribution extends Contribution{

    private String membership;

    public PhilHealthContribution(double monthlySalary, String membership) throws Exception {
        super(monthlySalary);
        setModel("PhilHealth");
        setMembership(membership);
        setMonthlyContribution(
            compute(monthlySalary, membership)
        );
    }

    public void setMembership(String membership) throws Exception{
        if (
            membership.equalsIgnoreCase("employed") ||
            membership.equalsIgnoreCase("self-employed")
        )
            this.membership = membership;

        else throw new Exception("Invalid membership.");
    }

    public String getMembership(){
        return membership;
    }

    public double compute(double monthlySalary, String membership){
        if ( membership.equalsIgnoreCase("employed") ) {
            if ( monthlySalary <= 10000 )
                return 137.50;
            else if ( monthlySalary >= 10000.01 && monthlySalary <= 39999.99 )
                return ( monthlySalary * 0.0275 ) / 2;
            else if ( monthlySalary >= 40000 )
                return 1100.0;
        }
        else if ( membership.equalsIgnoreCase("self-employed") ){
            if ( monthlySalary <= 25000 )
                return 200.0;
            else if ( monthlySalary > 25000 )
                return 300.0;
        }
        return 0.0;
    }
}
